sdk/storage/Azure.Storage.Blobs.ChangeFeed/src/BlobChangeFeedExtensions.cs

```diff
@@ -0,0 +1,168 @@
+
+namespace Azure.Storage.Blobs.ChangeFeed
+{
+    /// <summary>
+    /// BlobChangeFeedExtensions.
+    /// </summary>
+    public static class BlobChangeFeedExtensions
+    {
+        /// <summary>
+        /// GetChangeFeedClient.
+        /// </summary>
+        /// <param name="serviceClient"></param>
+        /// <returns><see cref="BlobChangeFeedClient"/>.</returns>
+        public static BlobChangeFeedClient GetChangeFeedClient(this BlobServiceClient serviceClient)
+        {
+            return new BlobChangeFeedClient(serviceClient);
+        }
+
+        /// <summary>
+        /// Builds a DateTimeOffset from a segment path.
+        /// </summary>
+        internal static DateTimeOffset? ToDateTimeOffset(this string segmentPath)
+        {
+            if (segmentPath == null)
+            {
+                return default;
+            }
+            string[] splitPath = segmentPath.Split(new char[] { '/' }, StringSplitOptions.RemoveEmptyEntries);
+
+            if (splitPath.Length < 3)
+            {
+                throw new ArgumentException($"{nameof(segmentPath)} is not a valid segment path.");
+            }
+
+            return new DateTimeOffset(
+                year: int.Parse(splitPath[2], CultureInfo.InvariantCulture),
+                month: splitPath.Length >= 4
+                    ? int.Parse(splitPath[3], CultureInfo.InvariantCulture)
+                    : 1,
+                day: splitPath.Length >= 5
+                    ? int.Parse(splitPath[4], CultureInfo.InvariantCulture)
+                    : 1,
+                hour: splitPath.Length >= 6
+                    ? int.Parse(splitPath[5], CultureInfo.InvariantCulture) / 100
+                    : 0,
+                minute: 0,
+                second: 0,
+                offset: TimeSpan.Zero);
+        }
+
+        /// <summary>
+        /// Rounds a DateTimeOffset down to the nearest hour.
+        /// </summary>
+        internal static DateTimeOffset? RoundDownToNearestHour(this DateTimeOffset? dateTimeOffset)
+        {
+            if (dateTimeOffset == null)
+            {
+                return null;
+            }
+
+            return new DateTimeOffset(
+                year: dateTimeOffset.Value.Year,
+                month: dateTimeOffset.Value.Month,
+                day: dateTimeOffset.Value.Day,
+                hour: dateTimeOffset.Value.Hour,
+                minute: 0,
+                second: 0,
+                offset: dateTimeOffset.Value.Offset);
+        }
+
+        /// <summary>
+        /// Rounds a DateTimeOffset up to the nearest hour.
+        /// </summary>
+        internal static DateTimeOffset? RoundUpToNearestHour(this DateTimeOffset? dateTimeOffset)
+        {
+            if (dateTimeOffset == null)
+            {
+                return null;
+            }
+
+            DateTimeOffset? newDateTimeOffest = dateTimeOffset.RoundDownToNearestHour();
+
+            return newDateTimeOffest.Value.AddHours(1);
```

Comment:
Is this right? If input is 15:00:00, it will return 16:00:00?

Reply:
Correct.

Reply:
I think this is a bug. We shouldn't add 1 hour to the endTime if it's exactly at hour.